scripts/test-utils/test-fixture-dir.ts

```diff
@@ -156,6 +159,53 @@ export function testFixtureDir<T extends TestFixtureConfig>(
                         Error.captureStackTrace(err, testFixtureDir);
                     }
 
+                    const isErrorSnapshot = outputName.startsWith('error');
+                    const isSuccessSnapshot = outputName.startsWith('expected');
+
+                    // If we change from a successful result to an error (or vice versa),
+                    // then either the snapshot or content for the failing file is empty,
+                    // and the diff printed in the error message isn't helpful. Here, we check for
+                    // that case, and then check if the other file has flipped the other way.
+                    // If it has, we throw a more helpful error message.
+                    if (isErrorSnapshot || isSuccessSnapshot) {
+                        const brokenResult = outputs[outputName];
+                        const brokenResultHasContent = Boolean(brokenResult);
+                        const brokenSnapshot = readFileSync(outputPath, 'utf8');
+                        const brokenSnapshotHasContent = Boolean(brokenSnapshot);
+                        if (brokenResultHasContent !== brokenSnapshotHasContent) {
+                            // This file flipped from content to empty, or vice versa
+                            const otherType = isErrorSnapshot ? 'expected' : 'error';
+                            const otherName = outputsList.find(([name]) =>
+                                name.startsWith(otherType)
+                            )![0];
+                            const otherResult = outputs[otherName];
+                            const otherResultHasContent = Boolean(otherResult);
+                            const otherSnapshot = readFileSync(
+                                path.resolve(dirname, otherName),
+                                'utf8'
+                            );
```

Comment:
It's a shame we have to keep reading in snapshot files when presumably Vitest is already doing this elsewhere.